lms/djangoapps/verify_student/tests/test_views.py

```diff
@@ -1940,73 +1943,148 @@ def setUp(self):
+        location_id = VerificationStatus.get_location_id(self.attempt)
+        usage_key = UsageKey.from_string(location_id)
+        redirect_url = get_redirect_url(self.course_key, usage_key.replace(course_key=self.course_key))
+        self.request = RequestFactory().get('/url')
+        self.course_link = self.request.build_absolute_uri(redirect_url)
 
     def test_approved_email_message(self):
-        """
-        Test email message for approved photo verification.
-        """
+
         subject, body = _compose_message_reverification_email(
-            self.course.id, self.user.id, self.reverification_location, "approved", True
+            self.course.id, self.user.id, self.reverification_location, "approved", self.request
         )
 
         self.assertIn(
-            "Your verification for course {course_name} and assessment {assessment} has been passed.".format(
-                course_name=self.course.display_name_with_default,
-                assessment=self.assessment
+            "We have successfully verified your identity for the {assessment} "
+            "assessment in the {course_name} course.".format(
+                assessment=self.assessment,
+                course_name=self.course.display_name_with_default
             ),
             body
         )
+
+        self.check_courseware_link_exists(body)
         self.assertIn("Re-verification Status", subject)
 
     def test_denied_email_message_with_valid_due_date_and_attempts_allowed(self):
 
-        __, body = _compose_message_reverification_email(
-            self.course.id, self.user.id, self.reverification_location, "denied", True
+        subject, body = _compose_message_reverification_email(
+            self.course.id, self.user.id, self.reverification_location, "denied", self.request
         )
 
         self.assertIn(
-            "Your verification for course {course_name} and assessment {assessment} has failed.".format(
+            "We could not verify your identity for the {assessment} assessment "
+            "in the {course_name} course. You have used "
+            "{used_attempts} out of {allowed_attempts} attempts to "
+            "verify your identity.".format(
                 course_name=self.course.display_name_with_default,
-                assessment=self.assessment
+                assessment=self.assessment,
+                used_attempts=1,
+                allowed_attempts=self.allowed_attempts + 1
             ),
             body
         )
 
-        self.assertIn("Assessment closes on {due_date}".format(due_date=get_default_time_display(self.due_date)), body)
-        self.assertIn("Click on link below to re-verify", body)
         self.assertIn(
-            "https://{}{}".format(
-                microsite.get_value('SITE_NAME', 'localhost'), self.re_verification_link
+            "You must verify your identity before the assessment "
+            "closes on {due_date}".format(
```

Comment:
Do we need a period after {due_date} and before the quotation marks to make sure there's a period at the end of the sentence (as in lines 1960 and 1980)? The e-mail looked fine but I just wanted to verify this.